### PR TITLE
Improving typing for form validation

### PR DESCRIPTION
**Pull request by DanielSchiavini**

### CODE REVIEW

File: src/components/form.tsx [hunk 1]
```diff
@@ -2,9 +2,10 @@ import * as React from 'react';
 import { Link } from 'react-router-dom';
 
 import { default as Button, ButtonStyle } from './button';
+import {TranslatedText} from '../utils/i18n';
 
 
-export type FormValidationResponse = { [key: string]: any[] | string }; // can be any[] if returned from I18n.format
+export type FormValidationResponse = { [key: string]: TranslatedText | FormValidationResponse };
```
Comment (woosie10) [inline src/components/form.tsx]: why can FormValidationResponse = FormValidationResponse?

Comment (ChrisTerBeke) [reply src/components/form.tsx]: Sometimes they are nested (for example the material spec sheet link pairs that Daniel is working on for the contribute portal.

**Comment by ChrisTerBeke** [comment]
Master should be merged into this to get the build to succeed again.

**Comment by DanielSchiavini** [comment]
@chris this is based on `make-inputs-controlled`

**Pull request merged**